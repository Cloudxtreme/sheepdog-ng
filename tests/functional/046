#!/bin/bash

# Test vdi deletion

. ./common

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format -c 2

echo "delete snapshot image before current one"
_vdi_create test 4G
$DOG vdi snapshot test -s snap
_vdi_list

$DOG vdi delete test -s snap
$DOG vdi delete test -s snap # error
_vdi_list

$DOG vdi delete test
$DOG vdi delete test # error
_vdi_list

echo "delete current image before snapshot one"
_vdi_create test 4G
$DOG vdi snapshot test -s snap
_vdi_list

$DOG vdi delete test
$DOG vdi delete test # error
_vdi_list

$DOG vdi delete test -s snap
$DOG vdi delete test -s snap # error
_vdi_list

# Test vdi deletio while in recovery

_vdi_create test 512M -P
$DOG vdi snapshot test -s snap

for i in `seq 0 49`; do
    echo $i | $DOG vdi write test $((i * 4 * 1024 * 1024)) 512
done

$DOG vdi clone test -s snap clone

for i in `seq 25 74`; do
    echo $i*2 | $DOG vdi write test $((i * 4 * 1024 * 1024)) 512
done

echo "delete snapshot"
$DOG vdi delete test -s snap &
_start_sheep 3
_wait_for_sheep 4
wait
_wait_for_sheep_recovery 0
_node_info

echo "delete test"
$DOG vdi delete test &
_kill_sheep 1
wait
_wait_for_sheep_recovery 0
_node_info | grep -v ^1

echo "delete clone"
$DOG vdi delete clone &
_start_sheep 4
_start_sheep 5
_wait_for_sheep 5
wait
_wait_for_sheep_recovery 0
_node_info | grep -v ^1

_vdi_list
