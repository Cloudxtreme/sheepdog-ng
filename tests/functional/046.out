QA output created by 046
using backend plain store
delete snapshot image before current one
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25      2          snap
  test         0  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26      2              
Failed to open VDI test (snapshot id: 0 snapshot tag: snap): Failed to find requested tag
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26      2              
Failed to open VDI test (snapshot id: 0 snapshot tag: ): No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
delete current image before snapshot one
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25      2          snap
  test         0  4.0 GB  0.0 MB  0.0 MB DATE   7c2b27      2              
Failed to open VDI test (snapshot id: 0 snapshot tag: ): No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25      2          snap
Failed to open VDI test (snapshot id: 0 snapshot tag: snap): No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
delete snapshot
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	83	2	0	0	62	0
1	110	1	0	0	61	0
2	108	5	0	0	61	0
3	105	2	0	0	72	0
delete test
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	78	2	0	0	80	0
2	87	5	0	0	80	0
3	91	3	0	0	96	0
delete clone
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	0	2	0	0	0	0
2	0	4	0	0	0	0
3	0	1	0	0	0	0
4	0	2	0	0	0	0
5	0	1	0	0	0	0
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
